src/token.rs

```diff
@@ -1,12 +1,73 @@
+use lazy_static::__Deref;
+
 /// This struct contains a token, an expiry, and an access scope.
 pub struct Token {
     // this field contains the JWT and the expiry thereof. They are in the same Option because if
     // one of them is `Some`, we require that the other be `Some` as well.
-    token: Option<(String, u64)>,
+    token: tokio::sync::RwLock<Option<(String, u64)>>,
     // store the access scope for later use if we need to refresh the token
     access_scope: String,
 }
 
+/// Trait that refreshes a token when it is expired
+#[async_trait::async_trait]
+pub trait RefreshableToken: Sync {
```

Comment:
Should we not call this trait `AuthCache` or `TokenCache` to better reflect what it does?